servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/DefaultBlockingIterableProcessor.java

```diff
@@ -144,43 +119,37 @@ public boolean hasNext() {
                 return true;
             }
 
-            final Object next;
+            final boolean consumed;
             try {
-                next = buffer.take();
+                consumed = buffer.consume(this);
             } catch (InterruptedException e) {
                 return throwException(e);
             }
-            return processHasNext(next);
+            return terminal == null ? consumed : hasNextWhenTerminated();
         }
 
-        @Nullable
-        private T processNext() {
-            Object next = this.next;
-            this.next = null;
-            if (next == NULL_MASK) {
-                return null;
-            } else {
-                @SuppressWarnings("unchecked")
-                T t = (T) next;
-                return t;
+        private boolean hasNextWhenTerminated() {
```

Comment:
how is the `boolean` useful here ? This always returns false....

Reply:
It just make it easy to use in `hasNext()`:

```java
return hasNextWhenTerminated()
```
instead of 
```java
hasNextWhenTerminated();
return false;
```